sdks/go/test/integration/io/xlang/bigquery/bigquery_test.go

```diff
@@ -0,0 +1,134 @@
+	// Number of random elements to create for test. Must be less than number of words in text.
+	inputSize = 50
+)
+
+func shuffleText() []string {
+	words := strings.Fields(text)
+	rand.Shuffle(len(words), func(i, j int) { words[i], words[j] = words[j], words[i] })
+	return words
+}
+
+// TestRow is a sample row to write and read from that is expected to contain enough deterministic
+// and random data in different data types to provide a reasonable signal that reading and writing
+// works at a basic level.
+type TestRow struct {
+	Counter   int64    `beam:"counter"`   // A deterministic counter, increments for each row generated.
+	Rand_data RandData `beam:"rand_data"` // An inner struct containing randomized data.
+}
+
+// RandData is a struct of various types of random data.
+type RandData struct {
+	Flip bool   `beam:"flip"` // Flip is a bool with a random chance of either result (a coin flip).
+	Num  int64  `beam:"num"`  // Num is a random int64.
+	Word string `beam:"word"` // Word is a randomly selected word from a sample text.
+}
+
+// ddlSchema is a string for BigQuery data definition language that corresponds to TestRow.
+const ddlTestRowSchema = "counter INT64 NOT NULL, " +
+	"rand_data STRUCT<" +
+	"flip BOOL NOT NULL," +
+	"num INT64 NOT NULL," +
+	"word STRING NOT NULL" +
+	"> NOT NULL"
+
+// createTestRows creates a number of TestRows, populating the randomized data.
+func createTestRows(_ []byte, emit func(TestRow)) {
+	rand.Seed(time.Now().UnixNano())
+	words := shuffleText()
+	for i := 0; i < inputSize; i++ {
+		emit(TestRow{
+			Counter: int64(i),
+			Rand_data: RandData{
+				Flip: rand.Int63n(2) != 0,
+				Num:  rand.Int63(),
+				Word: words[i],
+			},
+		})
+	}
+}
+
+// TestBigQueryIO_BasicWriteRead runs a pipeline that generates semi-randomized elements, writes
+// them to a BigQuery table and then reads from that table, and checks that the result matches the
+// original inputs. This requires a pre-existing table to be created.
+func TestBigQueryIO_BasicWriteRead(t *testing.T) {
+	integration.CheckFilters(t)
+	checkFlags(t)
+
+	// Create a table before running the pipeline
+	table, err := newTempTable(*integration.BigQueryDataset, "go_bqio_it", ddlTestRowSchema)
```

Comment:
consider printing the table name using `t.Log`, to make it clearer which resources map to this test.